test/test_functionalization.py

```diff
@@ -1542,6 +1543,87 @@ def forward(self, x_1):
     add = torch.ops.aten.add.Tensor(view_1, 1);  view_1 = None
     return add""")
 
+    def test_python_functionalization_zero_tensor(self):
+        def f(x):
+            y = torch.ops.aten._efficientzerotensor([4])
+            out = x + y
+            out.mul_(2)
+            return out
+        x = torch.randn(4)
+        out_ref = f(x)
+        out_test = dispatch_functionalize(f)(x)
+        out_test_cpp = _functionalize(f, reapply_views=True, crossref=False, skip_input_mutations=True)(x)
+        self.assertEqual(out_ref, out_test)
+        self.assertEqual(out_ref, out_test_cpp)
+        if not IS_WINDOWS:
+            fx_g = make_fx(dispatch_functionalize(f))(x)
+            fx_g_cpp = make_fx(_functionalize(f, reapply_views=True, crossref=False, skip_input_mutations=True))(x)
+            self.assertExpectedInline(fx_g.code.strip(), """\
+def forward(self, arg0_1):
+    _efficientzerotensor = torch.ops.aten._efficientzerotensor.default([4])
+    _to_copy = torch.ops.aten._to_copy.default(arg0_1, device = device(type='meta'), pin_memory = False)
+    zeros = torch.ops.aten.zeros.default([], dtype = torch.float32, layout = torch.strided, device = device(type='cpu'))
+    expand = torch.ops.aten.expand.default(zeros, [4]);  zeros = None
+    _to_copy_1 = torch.ops.aten._to_copy.default(expand, device = device(type='meta'), pin_memory = False);  expand = None
+    add = torch.ops.prims.add.default(_to_copy, _to_copy_1);  _to_copy = _to_copy_1 = None
```

Comment:
Why do we get torch.ops.prims in the trace?

Reply:
I had to double check, this is a good q. It looks like it's because ZeroTensor implements most binary ops by redispatching to the op with meta tensors ([code](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/BinaryOps.cpp#L1019)).

Since this is a direct redispatch, we go **straight** to the meta implementation of add/mul/sub, without hitting the python key first.

This invokes the meta implementation of `add()`, which we currently register from python, and calls the primtorch implementation ([code](https://github.com/pytorch/pytorch/blob/main/torch/_refs/__init__.py#L1080)).

I think in theory, we'd probably want any meta-compute that zero-tensor does to be const-propp'd and not bake the ops into the graph (future fix?).